### ISSUE TITLE
Hosting Multiple Stores/Websites with Nginx

### ISSUE THREAD

**Issue by camdixon** [magento/devdocs#1943]
## This issue is a:
[ ] Bug on the website
[ ] New topic request
[ ] Topic clarification request
[ ] New DevDocs feature request
[ X ] Other

## Description:
This dev docs page is meant to help Nginx users understand how to host with multiple websites and store views within a Magento 2 install.
https://devdocs.magento.com/guides/v2.2/config-guide/multi-site/ms_nginx.html#ms-nginx-vhosts

A) In Step 2 - "To Create virtual hosts" should be opened by default as it is the main point of this page.  It's the first portion of Nginx code someone will enter to activate a site, and therefore should not be hidden by default.

My coworkers and I also does not create the extra steps of having multiple virtual hosts as you can do this with just 1 vhosts file in Nginx, and a file in locations.d (Use Magento provided one with the additional of current page Step 3 for multi stores).

```
map $http_host $MAGE_RUN_CODE {
 default '';
 subomain1.snowboxes.com additional_website_code_one; 
 subomain2.snowboxes.com additional_website_code_two; 
 subomain3.snowboxes.com additional_website_code_three; 
 subomain4.snowboxes.com additional_website_code_four;
}
server {
    listen  80;
```

This is more efficient provided other items are not needed such as different PHP sockets per website.  If changes such as this or variations in access for specific urls are needed then in those specific cases additional vhosts are needed.

B) I also think this tip could be reworded and moved to the top of "Step 3".  
Do not edit the nginx.conf.sample file; it is a core Magento file that may be updated with each new release. Instead, copy the nginx.conf.sample file, rename it, and then edit the copied file.

C) Also, I was not sure if I should create a dev doc issue, and then do PR or just do a PR as how Magento expects this to happen.  Do you usually want open issue so team can decided if ok to do?

**Comment by shrielenee** [comment]
Hey @camdixon - if you'd like to create a PR that would be awesome! If you do that, we can close or link this issue to it. Let me know what you're thinking 👍

**Comment by camdixon** [comment]
@shrielenee I will create PR for it, but as to why I opened issue see item C :)

**Comment by jeff-matthews** [comment]
Hey @camdixon! Creating an issue first was the right choice. We want to encourage discussion before you start working on a PR to make sure no one is duplicating effort. See our contribution guidelines for more info: https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing_docs.html#contribution-guidelines

**Comment by jeff-matthews** [comment]
Also, we have some multi-website setup docs here as well: https://devdocs.magento.com/guides/v2.2/config-guide/multi-site/ms_over.html

I just want to be sure you're aware of everything before getting started.

**Comment by camdixon** [comment]
@jeff-matthews thanks for the links!   I guess on the Multi Site Overview I have a question about the page.
1) Configure Magento websites and stores
This section discusses the minimum tasks required to use the MAGE_RUN_TYPE and MAGE_RUN_CODE variables. (top of page)

Question: I don't think my improvements are "minimum tasks required", however unless someone is trying to allow a specific link through on a particular website I don't see the need for additional vhosts. My improvement is not the minimum per say, but it is actually less code, and much easier to maintain multi-store instances.  I also don't believe that the improvements of "map" are difficult for someone clicking on Nginx page for multi-stores to understand and implement. 

2) Create one virtual host per Magento website or store view. (bottom of page)
Creating another vhosts is great for 2 or 3 stores, but using a map is ideal for more than this in my experience.  You could also argue that Ansible or Puppet can handle all of these vhosts, and while this is also true I actually use the map function in our version of Ansible that we use to host Magento2 Multi Stores.  Even here, it's easier to maintain and works just as fast.

**Comment by camdixon** [comment]
@jeff-matthews @shrielenee Do you want me to add the updates to these 2 pages?

**Comment by jeff-matthews** [comment]
Yes @camdixon, please create a PR if you think you can improve the documentation. We may ask an internal developer to validate your changes though.

**Comment by camdixon** [comment]
@jeff-matthews please see PR #1953.

**Comment by jeff-matthews** [comment]
Thanks @camdixon. @shrielenee will review.
